packages/scandipwa/src/util/ContainerProps/ContainerProps.js

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable import/prefer-default-export */
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+export const getContainerProps = (componentOrIncludedPropsArray, ...props) => {
+    const allPropsObject = props.reduce((acc, prop) => ({
+        ...acc,
+        ...prop
+    }), {});
+
+    const propsFilter = (prop) => Object.prototype.hasOwnProperty.call(allPropsObject, prop);
+    const propsReducer = (acc, val) => ({
+        ...acc,
+        [val]: allPropsObject[val]
+    });
+
+    if (Array.isArray(componentOrIncludedPropsArray)) {
+        return componentOrIncludedPropsArray
+            .filter(propsFilter)
+            .reduce(propsReducer, {});
+    }
+
+    const { propTypes: requiredComponentProps = {} } = componentOrIncludedPropsArray || {};
```

Comment:
I guess the eslint is complaining about this rule [react/forbid-foreign-prop-types](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-foreign-prop-types.md).

Did you check the build?
Do we strip propTypes from the component?
Should we define propTypes in a separate variable and export it then?

Reply:
Yeah, it complains because it thinks that we're reusing propTypes between components, while actually, it's not true.
Probably it just checks for keyword `propTypes` outside of the class body and then just warns you.

Yes, we can define an array of props, or make a function that will return props names from the component class body. Then it will be easy to extend, and manage.

Just the thing is, at this moment, if we use example implementation, it will require very few changes in the codebase to make it work.